iac-extensions/arm/src/main/java/org/sonar/iac/arm/checks/SubscriptionRoleAssignmentCheck.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.arm.checks;
+
+import java.util.Map;
+import org.sonar.check.Rule;
+import org.sonar.iac.arm.checkdsl.ContextualResource;
+import org.sonar.iac.arm.tree.api.File;
+import org.sonar.iac.common.api.checks.SecondaryLocation;
+
+@Rule(key = "S6387")
+public class SubscriptionRoleAssignmentCheck extends AbstractArmResourceCheck {
+
+  private static final String MESSAGE = "Make sure assigning this role with a %s scope is safe here.";
+  private static final Map<File.Scope, String> SENSITIVE_SCOPE_WITH_NAME = Map.of(
```

Comment:
IMO a static import of Scope is also fine since it is not used in another context. I will make it more readable.